tests/end-to-end/external-solution-integration/pkg/testsuite/deploy_function.yaml.go

```diff
@@ -0,0 +1,198 @@
+	console.log("Received event: ");	
+	console.log(JSON.stringify(event, null, 2));	
+	console.log("==============================");	
+	console.log("Received context:");	
+	console.log(JSON.stringify(context, null, 2));	
+	console.log("==============================");	
+	
+	const gateway = getGateway()
+    if (gateway === undefined) {
+		throw new Error("gateway is undefined")	
+	}
+    if (event["data"] !== expectedPayload) {	
+		throw new Error("Payload not as expected")	
+    }
+	return new Promise((resolve, reject) => {	
+		const url = gateway + "/counter";	
+		console.log("Counter URL: ", url);	
+		sendReq(url, resolve, reject);	
+	}).then(resolved, rejected);	
+} };	
+`
+const functionDeps = `{"dependencies":{"request": "^2.88.0", "circular-json": "^0.5.9"}}`
+
+// DeployFunction deploys function to the cluster. The function will do PUT /counter to connected application upon receiving
+// an event
+type DeployFunction struct {
+	client          dynamic.ResourceInterface
+	name            string
+	port            int
+	expectedPayload string
+	legacy          bool
+}
+
+var _ step.Step = &DeployFunction{}
+
+func NewDeployFunction(name, expectedPayload string, port int, client dynamic.ResourceInterface, legacy bool) *DeployFunction {
+	return &DeployFunction{
+		client:          client,
+		name:            name,
+		port:            port,
+		legacy:          legacy,
+		expectedPayload: expectedPayload,
+	}
+}
+
+func (s *DeployFunction) Name() string {
+	return fmt.Sprintf("Deploy function %s", s.name)
+}
+
+func (s *DeployFunction) Run() error {
+	if err := s.createFunction(); err != nil {
+		return err
+	}
+
+	if err := retry.Do(s.isFunctionReady); err != nil {
+		return errors.Wrap(err, "function not ready")
+	}
+
+	return nil
+}
+
+func (s *DeployFunction) createFunction() error {
+	function := &serverless.Function{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Function",
+			APIVersion: serverless.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: s.name,
+		},
+		Spec: serverless.FunctionSpec{
+			Source: fmt.Sprintf(functionFmt, s.expectedPayload, s.legacy),
+			Deps:   functionDeps,
+		},
+	}
+
+	unstructuredObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(function)
+	if err != nil {
+		return err
+	}
+
+	_, err = s.client.Create(&unstructured.Unstructured{Object: unstructuredObj}, metav1.CreateOptions{})
+	return err
+}
+
+func (s *DeployFunction) Cleanup() error {
+	err := s.client.Delete(s.name, &metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	return retry.Do(
+		s.isLambdaTerminated)
+}
+
+func (s *DeployFunction) isFunctionReady() error {
+	function, err := s.getFunctions()
+	if err != nil {
+		return err
+	}
+
+	if function.Name == "" {
+		return errors.New("no function found")
+	}
+
+	if !s.isConditionReady(function) {
+		return errors.New("function is not ready yet")
+	}
+
+	return nil
+}
+
+func (s *DeployFunction) isLambdaTerminated() error {
```

Comment:
shouldnt this return a `bool`

also: the why do we use the word lambda here?

Reply:
My mistake, I forgot to change it. I will change it to "isFunctionTerminated"